`test_filecallback` can fail due to terminal characters in output

### Checked other resources

- [x] This is a bug, not a usage question. For questions, please use the LangChain Forum (https://forum.langchain.com/).
- [x] I added a clear and descriptive title that summarizes this issue.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I am sure that this is a bug in LangChain rather than my code.
- [x] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).
- [x] I read what a minimal reproducible example is (https://stackoverflow.com/help/minimal-reproducible-example).
- [x] I posted a self-contained, minimal, reproducible example. A maintainer can copy it and run it AS IS.

### Example Code

n/a While building for nixpkgs

### Error Message and Stack Trace (if applicable)

______________________________ test_filecallback _______________________________

capsys = <_pytest.capture.CaptureFixture object at 0x16537b4d0>
tmp_path = PosixPath('/private/tmp/nix-build-python3.13-langchain-0.3.26.drv-0/pytest-of-_nixbld1/pytest-0/test_filecallback0')

    def test_filecallback(capsys: pytest.CaptureFixture, tmp_path: pathlib.Path) -> Any:
        """Test the file callback handler."""
        p = tmp_path / "output.log"
        handler = FileCallbackHandler(str(p))
        chain_test = FakeChain(callbacks=[handler])
        chain_test.invoke({"foo": "bar"})
        # Assert the output is as expected
>       assert p.read_text() == (
            "\n\n\x1b[1m> Entering new FakeChain "
            "chain...\x1b[0m\n\n\x1b[1m> Finished chain.\x1b[0m\n"
        )
E       AssertionError: assert '\n\n> Entering new FakeChain chain...\n\n> Finished chain.\n' == '\n\n\x1b[1m> Entering new FakeChain chain...\x1b[0m\n\n\x1b[1m> Finished chain.\x1b[0m\n'
E         
E           
E           
E         - > Entering new FakeChain chain...
E         ? ----                                 ----
E         + > Entering new FakeChain chain...
E           
E         - > Finished chain.
E         ? ----                 ----
E         + > Finished chain.

tests/unit_tests/callbacks/test_file.py:43: AssertionError

=========================== short test summary info ============================
FAILED tests/unit_tests/callbacks/test_file.py::test_filecallback - AssertionError: assert '\n\n> Entering new FakeChain chain...\n\n> Finished chain.\n' == '\n\n\x1b[1m> Entering new FakeChain chain...\x1b[0m\n\n\x1b[1m> Finished chain.\x1b[0m\n'


### Description

While building with a low battery (zsh puts a low battery symbol before each line out), the test failed.

ANSI terminal control sequences should be stripped before comparison.

### System Info

langchain 0.3.26

--- a/libs/langchain/tests/unit_tests/callbacks/test_file.py
+++ b/libs/langchain/tests/unit_tests/callbacks/test_file.py
@@ -1,4 +1,5 @@
 import pathlib
+import re
 from typing import Optional
 
 from langchain_core.callbacks import CallbackManagerForChainRun
@@ -32,6 +33,17 @@ class FakeChain(Chain):
         return {"bar": "bar"}
 
 
+def strip_ansi(text: str) -> str:
+    """
+    Removes ANSI escape sequences from a string.
+
+    Args:
+        text: The string potentially containing ANSI codes.
+    """
+    ansi_escape = re.compile(r'\x1B\[[0-?]*[ -/]*[@-~]')
+    return ansi_escape.sub('', text)
+
+
 def test_filecallback(tmp_path: pathlib.Path) -> None:
     """Test the file callback handler."""
     log1 = tmp_path / "output.log"
@@ -40,7 +52,7 @@ def test_filecallback(tmp_path: pathlib.Path) -> None:
     chain_test.invoke({"foo": "bar"})
     handler.close()
     # Assert the output is as expected
-    assert "Entering new FakeChain chain" in log1.read_text()
+    assert "Entering new FakeChain chain" in strip_ansi(log1.read_text())
 
     # Test using a callback manager
     log2 = tmp_path / "output2.log"
@@ -49,12 +61,11 @@ def test_filecallback(tmp_path: pathlib.Path) -> None:
         chain_test = FakeChain(callbacks=[handler_cm])
         chain_test.invoke({"foo": "bar"})
 
-    assert "Entering new FakeChain chain" in log2.read_text()
+    assert "Entering new FakeChain chain" in strip_ansi(log2.read_text())
 
     # Test passing via invoke callbacks
-
     log3 = tmp_path / "output3.log"
 
     with FileCallbackHandler(str(log3)) as handler_cm:
         chain_test.invoke({"foo": "bar"}, {"callbacks": [handler_cm]})
-    assert "Entering new FakeChain chain" in log3.read_text()
+    assert "Entering new FakeChain chain" in strip_ansi(log3.read_text())

--- a/libs/langchain/tests/unit_tests/callbacks/test_file.py
+++ b/libs/langchain/tests/unit_tests/callbacks/test_file.py
@@ -40,8 +40,8 @@ def strip_ansi(text: str) -> str:
     Args:
         text: The string potentially containing ANSI codes.
     """
-    ansi_escape = re.compile(r'\x1B\[[0-?]*[ -/]*[@-~]')
-    return ansi_escape.sub('', text)
+    ansi_escape = re.compile(r"\x1B\[[0-?]*[ -/]*[@-~]")
+    return ansi_escape.sub("", text)
 
 
 def test_filecallback(tmp_path: pathlib.Path) -> None:

